src/interactive/commands.sml

```diff
@@ -66,14 +66,15 @@ struct
             INL s => raise Fail s
           | INR sign =>
             let
-              val elab = RefineElab.transport o ValidationElab.transport o BindSignatureElab.transport
-              val sign' = elab sign
+              val elab = ValidationElab.transport o BindSignatureElab.transport
+              val sign' = RefineElab.transport (AbtSignature.Telescope.append (oldSign, elab sign))
```

Comment:
Also, keep in mind that the signature elaboration architecture is not set in stone, and we can change it in whatever way would best enable what you're trying to do 😄